go/service/chat_local.go

```diff
@@ -126,251 +207,141 @@ func (h *chatLocalHandler) GetThreadLocal(ctx context.Context, arg chat1.GetThre
-		return chat1.NewConversationLocalRes{}, fmt.Errorf("error getting crypt keys %s", err)
-	}
-	tlfIDb := res.TlfID.ToBytes()
-	if tlfIDb == nil {
-		return chat1.NewConversationLocalRes{}, errors.New("invalid TlfID acquired")
+		return chat1.NewConversationLocalRes{}, err
 	}
-	tlfID := chat1.TLFID(tlfIDb)
 
-	info.Triple = chat1.ConversationIDTriple{
+	triple := chat1.ConversationIDTriple{
 		Tlfid:     tlfID,
-		TopicType: info.TopicType,
+		TopicType: arg.TopicType,
 		TopicID:   make(chat1.TopicID, 16),
 	}
-	info.TlfName = string(res.CanonicalName)
 
 	for i := 0; i < 3; i++ {
-		if info.Triple.TopicType != chat1.TopicType_CHAT {
+		if triple.TopicType != chat1.TopicType_CHAT {
 			// We only set topic ID if it's not CHAT. We are supporting only one
 			// conversation per TLF now. A topic ID of 0s is intentional as it would
 			// cause insertion failure in database.
 
-			if info.Triple.TopicID, err = libkb.NewChatTopicID(); err != nil {
+			if triple.TopicID, err = libkb.NewChatTopicID(); err != nil {
 				return chat1.NewConversationLocalRes{}, fmt.Errorf("error creating topic ID: %s", err)
 			}
 		}
 
-		firstMessageBoxed, err := h.prepareMessageForRemote(ctx, makeFirstMessage(ctx, info))
+		firstMessageBoxed, err := h.makeFirstMessage(ctx, triple, cname, arg.TlfVisibility, arg.TopicName)
 		if err != nil {
 			return chat1.NewConversationLocalRes{}, fmt.Errorf("error preparing message: %s", err)
 		}
 
-		var res chat1.NewConversationRemoteRes
-		res, err = h.remoteClient().NewConversationRemote2(ctx, chat1.NewConversationRemote2Arg{
-			IdTriple:   info.Triple,
+		var ncrres chat1.NewConversationRemoteRes
+		ncrres, reserr = h.remoteClient().NewConversationRemote2(ctx, chat1.NewConversationRemote2Arg{
+			IdTriple:   triple,
 			TLFMessage: *firstMessageBoxed,
 		})
-		fres.RateLimits = h.aggRateLimits([]*chat1.RateLimit{res.RateLimit})
-		if err != nil {
-			if cerr, ok := err.(libkb.ChatConvExistsError); ok {
-				if info.Triple.TopicType == chat1.TopicType_CHAT {
-					// A chat conversation already exists; just reuse it.
-					info.Id = cerr.ConvID
-					fres.Conv = info
-					return fres, nil
+		if ncrres.RateLimit != nil {
+			res.RateLimits = append(res.RateLimits, *ncrres.RateLimit)
+		}
+		convID := ncrres.ConvID
+		if reserr != nil {
+			if cerr, ok := reserr.(libkb.ChatConvExistsError); ok {
+				if triple.TopicType != chat1.TopicType_CHAT {
+					// Not a chat conversation. Multiples are fine. Just retry with a
+					// different topic ID.
+					continue
 				}
-
-				// Not a chat conversation. Multiples are fine. Just retry with a
-				// different topic ID.
-				continue
+				// A chat conversation already exists; just reuse it.
+				convID = cerr.ConvID
 			}
 		}
 
-		info.Id = res.ConvID
-		fres.Conv = info
+		// create succeeded; grabbing the conversation and returning
 
-		return fres, nil
-	}
-
-	return chat1.NewConversationLocalRes{}, err
-}
+		gilres, err := h.GetInboxLocal(ctx, chat1.GetInboxLocalArg{
```

Comment:
Maybe the server should just return the whole conversation, maybe just make a ticket for that.